### PR TITLE
[docs] add MTP guide

### PR DESCRIPTION
**Pull request by cuichenx**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive guide for Multi-Token Prediction (MTP) training covering configuration parameters, usage examples, parameter tuning guidelines, loss monitoring techniques, pipeline parallelism considerations, and troubleshooting solutions.
  * Updated documentation structure to include the new MTP training guide in both the Training and Customization section and main Guides list.

### CODE REVIEW

File: docs/training/multi-token-prediction.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+### Minimal Configuration Example
+
+Here's a minimal example using the Qwen3 30B-A3B recipe with MTP enabled:
+
+```python
+from megatron.bridge.recipes.qwen import qwen3_30b_a3b_pretrain
+from megatron.bridge.training.pretrain import pretrain
+
+config = qwen3_30b_a3b_pretrain(
```
Comment (yaoyu-33) [inline docs/training/multi-token-prediction.md]: can we use the new refactored way setting things.
config = qwen3_30b_a3b_pretrain()
config.data.data_paths=[...]
...

File: docs/training/multi-token-prediction.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+When using Pipeline Parallelism (PP), **MTP layers must be placed in the last pipeline stage** alongside the loss computation layer. Configure this using custom pipeline layout settings (`pipeline_model_parallel_split_rank`).
+
+### Pipeline Layout Guidelines
+
+MTP layers take approximately the same training time as a regular transformer layer. When configuring your pipeline layout:
+
+- **Place MTP in the last PP stage** (required for correct loss computation)
+- **Reduce layers in other PP ranks** to balance computation time across stages
+- Example: For a 60-layer model with PP=4 and `mtp_num_layers=1`, you might use splits like `[15, 30, 45, 60]` instead of `[15, 30, 46, 60]` to account for MTP overhead in the last stage
```
Comment (yaoyu-33) [inline docs/training/multi-token-prediction.md]: I dont understand the splits, are these accumulated values?

Comment (cuichenx) [reply docs/training/multi-token-prediction.md]: fixed now

File: docs/training/multi-token-prediction.md (hunk 1 above)
Comment (cuichenx) [reply docs/training/multi-token-prediction.md]: done
